dll/win32/iernonce/registry.cpp

```diff
@@ -199,14 +201,25 @@ BOOL RunOnceExSection::CloseAndDelete(
     return hParentKey.RecurseDeleteKey(m_SectionName) == ERROR_SUCCESS;
 }
 
-BOOL RunOnceExSection::Exec()
+UINT RunOnceExSection::GetEntryCnt() const
+{
+    return m_EntryList.GetSize();
+}
+
+BOOL RunOnceExSection::Exec(
+    _Inout_ UINT &iCompleteCnt,
+    _In_ const UINT iTotalCnt)
 {
     BOOL bSuccess = TRUE;
 
     for (int i = 0; i < m_EntryList.GetSize(); i++)
     {
         m_EntryList[i].Delete(m_RegKey);
         bSuccess &= m_EntryList[i].Exec();
+        iCompleteCnt++;
+        // TODO: the meaning of the third param is still unknown, seems it's always 0.
+        if (g_Callback)
+            g_Callback(iCompleteCnt, iTotalCnt, NULL);
```

Comment:
If the third parameter is always 0 in all cases and it has no effect whatsoever on how RunOnceEx works as per your research then a reasonable explanation for this is that this parameter is reserved for later usage in the next iteration of Windows, presumably starting with Vista or whatever, therefor it's left unused in the Server 2003 (NT 5.2) version of RunOnceEx.

In this case I don't think it's worth investing much time for this and can be investigated later post-GSoC since it doesn't bother us, if one wants.

Reply:
Of course I've thought about this situation and tested too, still 0 🤔